moztelemetry/hll.py

```diff
@@ -0,0 +1,72 @@
+"""
+Original code from spark-hyperloglog
+
+import com.mozilla.spark.sql.hyperloglog.aggregates._
+import com.mozilla.spark.sql.hyperloglog.functions._
+
+val hllMerge = new HyperLogLogMerge
+sqlContext.udf.register("hll_merge", hllMerge)
+sqlContext.udf.register("hll_create", hllCreate _)
+sqlContext.udf.register("hll_cardinality", hllCardinality _)
+
+val frame = sc.parallelize(List("a", "b", "c", "c")).toDF("id")
+val count = frame
+  .select(expr("hll_create(id, 12) as hll"))
+  .groupBy()
+  .agg(expr("hll_cardinality(hll_merge(hll)) as count"))
+  .show()
+
+
+
+The interface to the spark-hyperloglog API
+
+from pyspark.sql.types import *
+from moztelemetry import hll
+
+hll.register(spark)
+
+schema = StructType([StructField('id', StringType(), True)])
+df = spark.createDataFrame([("a",), ("b",), ("c",), ("c",)], ['id'], schema)
+count = df
+    .select(hll.create('id', 12).alias('hll'))
+    .agg(hll.cardinality(hll.merge('hll')).alias('count'))
+    .show()
+
+"""
+from pyspark.sql import SparkSession
+from pyspark.sql.column import Column, _to_java_column, _to_seq
+from pyspark.sql.types import BinaryType, LongType
+
+
+def hyperloglog():
+    session = SparkSession.builder.getOrCreate()
+    return session.sparkContext._jvm.com.mozilla.spark.sql.hyperloglog
+
+
+def register(session):
+    session.catalog.registerFunction(
+        'hll_merge', merge, returnType=BinaryType()
+    )
+    session.catalog.registerFunction(
+        'hll_cardinality', cardinality, returnType=LongType()
+    )
+    session.catalog.registerFunction(
+        'hll_create', create, returnType=BinaryType(),
```

Comment:
Same for the create function, `BinaryType` is probably the right one, as it [returns that](https://github.com/vitillo/spark-hyperloglog/blob/7a30b2efd7f4fa8c0e0f6e595ec600c37bb4fb19/src/main/scala/com/mozilla/spark/sql/hyperloglog/functions/package.scala#L25-L28).